motoman_mpx3500_support/urdf/mpx3500_macro.xacro

```diff
@@ -0,0 +1,170 @@
+        </visual>
+        <collision>
+            <geometry>
+                <mesh filename="package://motoman_mpx3500_support/meshes/mpx3500/collision/link_5_b.stl"/>
+            </geometry>
+        </collision>
+    </link>
+    <link name="${prefix}link_6_t">
+        <visual>
+            <geometry>
+                <mesh filename="package://motoman_mpx3500_support/meshes/mpx3500/visual/link_6_t.stl"/>
+            </geometry>
+            <material name="yaskawa_silver"/>
+        </visual>
+        <collision>
+            <geometry>
+                <mesh filename="package://motoman_mpx3500_support/meshes/mpx3500/collision/link_6_t.stl"/>
+            </geometry>
+        </collision>
+    </link>
+    <link name="${prefix}tool0"/>
+    <!-- end of link list -->
+
+    <!-- joint list -->
+    <joint name="${prefix}joint_1_s" type="revolute">
+        <parent link="${prefix}base_link"/>
+        <child link="${prefix}link_1_s"/>
+        <origin xyz="0 0 0.800" rpy="0 0 0" />
+        <axis xyz="0 0 1" />
+        <limit lower="${radians(-150)}" upper="${radians(150)}" effort="5516" velocity="${radians(100)}"/>
+    </joint>
+    <joint name="${prefix}joint_2_l" type="revolute">
+        <parent link="${prefix}link_1_s"/>
+        <child link="${prefix}link_2_l"/>
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <axis xyz="0 1 0" />
+        <limit lower="${radians(-65)}" upper="${radians(140)}" effort="7060" velocity="${radians(100)}"/>
+    </joint>
+    <joint name="${prefix}joint_3_u" type="revolute">
+        <parent link="${prefix}link_2_l"/>
+        <child link="${prefix}link_3_u"/>
+        <origin xyz="0 0 1.3" rpy="0 0 0" />
+        <axis xyz="0 -1 0" />
+        <limit lower="${radians(-65)}" upper="${radians(90)}" effort="3089" velocity="${radians(110)}"/>
+    </joint>
+    <joint name="${prefix}joint_4_r" type="revolute">
+        <parent link="${prefix}link_3_u"/>
+        <child link="${prefix}link_4_r"/>
+        <origin xyz="1.4 0 0" rpy="0 0 0" />
+        <axis xyz="-1 0 0" />
+        <limit lower="${radians(-720)}" upper="${radians(720)}" effort="926" velocity="${radians(300)}"/>
+    </joint>
+    <joint name="${prefix}joint_5_b" type="revolute">
+        <parent link="${prefix}link_4_r"/>
+        <child link="${prefix}link_5_b"/>
+        <origin xyz="0 0 0" rpy="0 ${radians(60)} 0" />
+        <axis xyz="-1 0 0" />
+        <limit lower="${radians(-720)}" upper="${radians(720)}" effort="794" velocity="${radians(360)}"/>
+    </joint>
+    <joint name="${prefix}joint_6_t" type="revolute">
+        <parent link="${prefix}link_5_b"/>
+        <child link="${prefix}link_6_t"/>
+        <origin xyz="0.114 0 0" rpy="0 ${radians(-60)} 0" />
+        <axis xyz="-1 0 0" />
+        <limit lower="${radians(-720)}" upper="${radians(720)}" effort="375" velocity="${radians(360)}"/>
+    </joint>
+    <joint name="${prefix}joint_6_t-tool0" type="fixed">
+        <origin xyz="0.123 0 0" rpy="3.1415926 -1.570796 0"/>
```

Comment:
I noticed some odd behavior with the LMA kinematics solver using the MoveIt planning GUI where the robot would jump backwards a few millimeters whenever the interactive marker on the TCP was clicked on but not dragged. Changing to the solution below seems to fix this issue; it might possibly have been caused by rounding

```suggestion
        <origin xyz="0.123 0 0" rpy="${radians(180)} ${radians(-90)} 0"/>
```